### ISSUE TITLE
Could the KRL Standard Library be written in KRL itself?

### ISSUE THREAD

**Issue by alexkolson** [Picolab/node-krl-stdlib#4]

**Comment by windley** [comment]
This is a good question to contemplate. What is the smallest set of primitives necessary to allow the rest of the standard library to be created from it? 

A few things that are harbingers of this:

1. The weather and other libraries from KRL's early period have been removed since they can be implemented in KRL with the HTTP library.  This made Dockerizing KRE easier. 

2. KRLv2 won't have Web actions built in on the assumption that these actions are able to be implemented with send_raw()

3. The [System Ruleset](http://www.windley.com/archives/2016/06/a_system_ruleset_for_the_pico_engine.shtml) shows one way the underlying engine can interface with KRL so that more of what might otherwise be custom code in the engine can be implemented as KRL once the engine has the proper hooks. 

In short, I like this idea and this line of thinking.

**Comment by 0joshuaolson1** [comment]
Note that [putting parent-child information in the engine](https://github.com/Picolab/pico-engine/issues/198) is one exception to this.

**Comment by 0joshuaolson1** [comment]
Copied to https://github.com/Picolab/wiki-issues/issues/43
